zebra-chain/src/orchard/tests/prop.rs

```diff
@@ -0,0 +1,35 @@
+use proptest::prelude::*;
+use std::io::Cursor;
+
+use crate::{
+    orchard,
+    serialization::{ZcashDeserializeInto, ZcashSerialize},
+};
+
+proptest! {
+    #[test]
+    // Make sure only valid flags deserialize
+    fn flags_roundtrip(flags in any::<u8>()) {
```

Comment:
We should probably also change the test name to make it clear that we're testing bytes, not structures:
```suggestion
    fn flag_roundtrip_bytes(flags in any::<u8>()) {
```